File: Algonrithm/Java/src/www/douyou/shanghai/www/douyou/shanghai/array/BubbleSort.java
package www.douyou.shanghai.www.douyou.shanghai.array;

import java.util.Arrays;

public class BubbleSort {
    public static void main(String[] args){
        int[] nums = {2,9,5,3,7};
        /**
         * 定义数组，冒泡排序的的中的是比较前后两个数字的大小
         * 每一次将最大或最小的数找到冒出到顶部或底部
         * 每次遍历数组内的元素变动
         */
        for (int i = 0; i <nums.length-1; i++) {
            for (int j = 0; j <nums.length-i-1; j++) {
                if(nums[j]<nums[j+1]){
                    int temp = nums[j];
                    nums[j] = nums[j+1];
                    nums[j+1] = temp;
                }
            }
        }
        System.out.println(Arrays.toString(nums));
    }
}

File: Test-c/master/ArraySort/BubbleSort.c
#include<stdio.h>

int main()
{
	int a[] = { 1,2,3,4,5,6,7,8,9 };
	int len = sizeof(a) / sizeof(a[0]);

	for (int end = len - 1; end>0; end--)
	{
		//第二个for是比较的次数
		for (int i = 0; i<end; i++)
			if (a[i] > a[i + 1]) {
				int temp;
				temp = a[i];
				a[i] = a[i + 1];
				a[i + 1] = temp;
			}
	}
	for (int i = 0; i < len; i++)
		printf("%d\t", a[i]);
	printf("%d\n", len);
	return 0;
}

void bubble_sort(int a[], int n)
{
	//将a中整数序列重新排列成自小至大
	//有序的整数序列
	for (i = n - 1, change = true; i > 1 && change; --i)
	{
		change = false;
		for (j = 0; j < i; ++j)
		{
			if (a[j] > a[j + 1]) {
				a[j]<---->a[j + 1]; change = true
			};
			
		}
	}
}